docs/spl-docs/rex-command.md

```diff
@@ -0,0 +1,104 @@
+# rex
+
+## Description
+
+Use this command to either extract fields using regular expression named groups, or replace or substitute characters in a field using sed expressions.
+
+The rex command matches the value of the specified field against the unanchored regular expression and extracts the named groups into fields of the corresponding names.
+
+## Syntax
+
+The required syntax is in **bold**.
+
+**rex** \
+**[field=\<field>]** \
+**\<regex-expression>**
+
+## Required Arguments
+
+### field
+
+**Syntax:** `field=<field>` \
+**Description:** The field that you want to extract information from.
+
+### regex-expression
+
+**Syntax:** `"<string>"` \
+**Description:** The PCRE regular expression that defines the information to match and extract from the specified field.
+
+### Examples
+
+### 1. Extract major and minor version number from an app version.
+
+This search extracts the major and minor version numbers from the `app_version` field using named groups and displays them as separate fields.
+
+```
+... | rex field=app_version "(?<major>\d+)\.(?<minor>\d).*"  
+| fields app_version, major, minor
+```
+
+**Example field value:**
+```
+app_version = "2.3.5" will be extracted as major = "2" and minor = "3"
+```
+
+### 2. Extract an email domain from an email address.
+
+This search extracts the domain part of the email addresses in the `user_email` field.
+
+```
+... | rex field=user_email "(?<domain>@\w+\.\w+)"
+| fields user_email, domain
+```
+
+**Example field value:**
+```
+user_email = "example@domain.com" will be extracted as domain = "domain.com"
+```
```

Comment:
nit: the extracted domain is `@domain.com`, not `domain.com`
<img width="890" alt="image" src="https://github.com/user-attachments/assets/69542ff9-59db-443f-b4fc-6e7004d90457">